### PR TITLE
Add `emisv2.practice_registrations`

### LINKED ISSUE #2786: Implement a `emisv2.practice_registrations` table
We would like to run a study against the EMISv2 backend that queries for patient's practice registrations.
This requires querying the `registration_start_datetime` and `registration_end_datetime` ~~and `organisation`~~ columns on the raw EMIS `patient` table. 

We should:
- [x] Ensure the relevant EMIS columns listed above are present in the [schema.py](https://github.com/opensafely-core/ehrql/blob/main/tests/backend_schemas/emisv2/schema.py) file and tested via integration tests
- [x] Implement a `emisv2.practice_registrations` query table with the following columns / methods:
  - [x] `start_date`
  - [x] `end_date`
  - [ ]  ~~`practice_pseudo_id`~~ See note below

We won't implement the `practice_pseudo_id` column for now as that work is potentially a bit involved and will be tracked in a separate issue. This is so that we can start trying to run the study, which does not use the column.

### PR DESCRIPTION
**Pull request by alarthast**
Closes #2786. This unblocks running a test study on the EMISv2 backend.

Similar to `emisv2.addresses`, we add `practice_registrations` as a event table even though it comes from the one-row-per-patient raw `patient` table in EMIS v2.

We do not implement the `practice_pseudo_id` column, and it is on the `core` table, so the EMIS `practice_registrations` has to be standalone.

Implementing `practice_pseudo_id` such that the table matches the core spec will be tackled separately in the future (see #2794).

### CODE REVIEW

File: ehrql/tables/emisv2.py [hunk 1]
```diff
@@ -74,3 +75,55 @@ def for_patient_on(self, date):
...
+    )
+
+    def for_patient_on(self, date):
+        """
+        Return each patient's practice registration as it was on the supplied date.
+        """
+        # Note that practice_registrations is an event-level table, but for EMISv2, it is
+        # derived from the patient table, so we know that there can only be at most one
+        # matching registration per patient
+        spanning_regs = self.where(self.start_date <= date).except_where(
```
Comment (rebkwok) [inline ehrql/tables/emisv2.py]: Why don't we just use `self.spanning(date, date)` here? I _think_ they do the same thing. (I know this is what the equivalent in the core tables does, but I'm not sure why we don't use it there either.)

File: tests/integration/tables/test_emisv2.py [hunk 2]
```diff
@@ -29,6 +29,204 @@ def test_patients_age_on(in_memory_engine):
...
+            emisv2.practice_registrations: [
+                dict(
+                    patient_id=4, start_date=date(2000, 1, 1), end_date=date(2012, 1, 1)
+                ),
+                dict(
+                    patient_id=4, start_date=date(2000, 1, 1), end_date=date(2015, 1, 1)
+                ),
+            ]
+        },
+    )
```
Comment (rebkwok) [inline tests/integration/tables/test_emisv2.py:58]: These tests aren't really representative of the real data, since we assume we only have one practice reg per patient. However, I think they're all logically OK apart from the last one; since we only sort the "spanning" registrations by start date, the "choose longest" one is only coming up with the right result accidentally, I think because end date is being used as a hidden tie-break for the sort. It's probably best to just remove that one, because it's misleading

File: tests/integration/tables/test_emisv2.py [hunk 3]
```diff
@@ -29,6 +29,204 @@ def test_patients_age_on(in_memory_engine):
     ]
 
 
+def test_practice_registrations_for_patient_on(in_memory_engine):
+    in_memory_engine.populate(
+        # Simple case: successive registrations
```
Comment (evansd) [inline tests/integration/tables/test_emisv2.py]: I'm not sure it makes sense to have these test cases given that they're not possible in the EMIS data. I'm worried they're a bit misleading and someone will end up thinking they must be there for a reason.

**Review by evansd — APPROVED** [review]
Looks good, just not sure about all the tests.

File: tests/integration/tables/test_emisv2.py (hunk 3 above)
Comment (alarthast) [reply tests/integration/tables/test_emisv2.py]: Thank you, I've hopefully now removed the misleading tests and added the comment about only having one registration, similar to how we do for addresses :+1:

File: tests/integration/tables/test_emisv2.py (hunk 2 above)
Comment (alarthast) [reply tests/integration/tables/test_emisv2.py:58]: Many thanks for catching that! This test has now been removed along with the other test cases that use more than one registration entry (see comment below as well) :+1:

File: ehrql/tables/emisv2.py (hunk 1 above)
Comment (alarthast) [reply ehrql/tables/emisv2.py]: [Discussed on Slack](https://bennettoxford.slack.com/archives/C080S7W2ZPX/p1778601647248559?thread_ts=1778591413.300969&cid=C080S7W2ZPX).

I've changed the definition of `spanning`:
https://github.com/opensafely-core/ehrql/blob/bf548b784d98e0c7f6e36c661c3ae340afe00bec/ehrql/tables/emisv2.py#L118-L126

and used it within `for_patient_on`. I've also tweaked the tests so that the edge cases are covered.

This means that the emisv2 implementation of the `spanning` helper method differs from the core one, but the amendments to the core method will be dealt with separately in #2690 .

**Review by rebkwok — APPROVED** [review]
